float32 transX
float32 transY
float32 transZ
float32 rotX
float32 rotY
float32 rotZ